workbench/test_utils.py

```diff
@@ -0,0 +1,144 @@
+
+    Args:
+        scenario_path (str): Path to the scenario XML file.
+
+    Keyword Arguments:
+        user_id (str or None): User ID to log in as, or None.
+
+    Returns:
+        The decorated method
+
+    Example:
+
+        @scenario('data/test_scenario.xml')
+        def test_submit(self, xblock):
+            response = self.request(xblock, 'submit', 'Test submission')
+            self.assertTrue('Success' in response)
+    """
+    def _decorator(func):
+        @wraps(func)
+        def _wrapped(*args, **kwargs):
+
+            # Retrieve the object (self)
+            # if this is a function, not a method, then do nothing.
+            xblock = None
+            if args:
+                self = args[0]
+                if isinstance(self, XBlockHandlerTestCaseMixin):
+
+                    # Print a debug message
+                    print "Loading scenario from {path}".format(path=scenario_path)
+
+                    # Configure the runtime with our user id
+                    self.set_user(user_id)
+
+                    # Load the scenario
+                    xblock = self.load_scenario(scenario_path)
+
+                    # Pass the XBlock as the first argument to the decorated method (after `self`)
+                    args = list(args)
+                    args.insert(1, xblock)
+
+            return func(*args, **kwargs)
+        return _wrapped
+    return _decorator
+
+
+class XBlockHandlerTestCaseMixin(object):
+    """Load the XBlock in the workbench runtime to test its handler. """
+
+    def setUp(self):
+        """Create the runtime. """
+        super(XBlockHandlerTestCaseMixin, self).setUp()
+        self.runtime = WorkbenchRuntime()
+
+    def set_user(self, user_id):
+        """Provide a user ID to the runtime.
+
+        Args:
+            user_id (str): a user ID.
+
+        Returns:
+            None
+        """
+        self.runtime.user_id = user_id
+
+    def load_scenario(self, xml_path):
+        """Load an XML definition of an XBlock and return the XBlock instance.
+
+        Args:
+            xml (string): Path to an XML definition of the XBlock, relative
+                to the test module.
+
+        Returns:
+            XBlock
+        """
+        block_id = self.runtime.parse_xml_string(
+            self.load_fixture_str(xml_path), self.runtime.id_generator
+        )
+        return self.runtime.get_block(block_id)
+
+    def request(self, xblock, handler_name, content, request_method="POST", response_format=None):
+        """Make a request to an XBlock handler.
+
+        Args:
+            xblock (XBlock): The XBlock instance that should handle the request.
+            handler_name (str): The name of the handler.
+            content (unicode): Content of the request.
+
+        Keyword Arguments:
+            request_method (str): The HTTP method of the request (defaults to POST)
+            response_format (None or str): Expected format of the response string.
```

Comment:
It seems like `response_format` is mainly here as a convenience for saving a call to json.loads.  It's not a sticking point for me, but for closer HTTP simulation we could rename this arg `content_type` and pass a proper content_type specification (e.g. "application/json") to `webob.Request`.  This would only be useful if blocks tend to make use of the `Accept` headers in handlers, and you want tests to be able to exercise that.